src/systems/a320_systems/src/hydraulic/flaps_computer.rs

```diff
@@ -0,0 +1,1393 @@
+                if actual_minus_target.get::<degree>().abs() > Self::ANGLE_DELTA_DEGREE {
+                    self.current_angle += Angle::new::<degree>(
+                        actual_minus_target.get::<degree>().signum()
+                            * self.speed.get::<degree_per_second>()
+                            * context.delta_as_secs_f64(),
+                    );
+                    self.current_angle = self.current_angle.max(Angle::new::<degree>(0.));
+                } else {
+                    self.current_angle = demanded_angle;
+                }
+            }
+        }
+    }
+}
+
+impl SimulationElement for SlatFlapGear {
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(
+            &self.left_position_percent_id,
+            self.current_angle / self.max_angle,
+        );
+        writer.write(
+            &self.right_position_percent_id,
+            self.current_angle / self.max_angle,
+        );
+        writer.write(&self.left_position_angle_id, self.current_angle);
+        writer.write(&self.right_position_angle_id, self.current_angle);
+    }
+}
+
+pub struct SlatFlapComplex {
+    sfcc: SlatFlapControlComputer,
+    flaps_handle: FlapsHandle,
+    flap_gear: SlatFlapGear,
+    slat_gear: SlatFlapGear,
+}
+
+impl SlatFlapComplex {
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            sfcc: SlatFlapControlComputer::new(context),
+            flaps_handle: FlapsHandle::new(context),
+            flap_gear: SlatFlapGear::new(
+                context,
+                AngularVelocity::new::<degree_per_second>(2.),
+                Angle::new::<degree>(40.),
+                "FLAPS",
+            ),
+            slat_gear: SlatFlapGear::new(
+                context,
+                AngularVelocity::new::<degree_per_second>(1.5),
+                Angle::new::<degree>(27.),
+                "SLATS",
+            ),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        hyd_green_pressure: Pressure,
+        hyd_blue_pressure: Pressure,
+        hyd_yellow_pressure: Pressure,
+    ) {
+        self.sfcc.update(
+            context,
+            &self.flaps_handle,
+            &self.flap_gear,
+            &self.slat_gear,
+        );
+        self.flap_gear
+            .update(context, &self.sfcc, hyd_green_pressure, hyd_yellow_pressure);
+        self.slat_gear
+            .update(context, &self.sfcc, hyd_green_pressure, hyd_blue_pressure);
+    }
+}
+
+impl SimulationElement for SlatFlapComplex {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        self.flaps_handle.accept(visitor);
+        self.sfcc.accept(visitor);
+        self.flap_gear.accept(visitor);
+        self.slat_gear.accept(visitor);
+        visitor.visit(self);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ntest::assert_about_eq;
```

Comment:
No hard need for this, I just feel checking an equality with (A-B).abs() < EPSION gets quite complicated to read at first glance when scrolling through tests.
You expect issues related with that test module?

Reply:
No, I just want to get some consensus on this. This is a big project, and I think we should just question pulling in external dependencies. They will need to be kept up to date, can potentially break things, and we eventually might end up with dependencies that duplicate functionality. So here I am, questioning if we really need this :-) (and the answer could be: yes!).

Should we in future flag code using epsilon and suggest to use the ntest call instead?

I'm also curious about @davidwalschots as well here, as together with you he's one of the other significant Rust contributors. David, is this library something you could also see using instead of epsilon comparisons going forwards?

Reply:
Might just be as simple as having any test module doing its own equality function so that it's written with the correct epsilon by the test writer.
fn is_equal_body_angle(angle1 angle2) ->bool;
fn is_equal_actuator_position(angle1 angle2) ->bool;